web/middleware/csrf.go

```diff
@@ -0,0 +1,45 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/bashmohandes/go-askme/web/framework"
+	"github.com/google/uuid"
+)
+
+// CSRF adds a csrf token to all sessions
+type csrf struct {
+}
+
+// Add CSRF token if it does not exist in session
+func (c *csrf) Run(cxt framework.Context) bool {
+	session := cxt.Session()
+	if session.Get("csrf") == nil {
+		session.Set("csrf", uuid.New().String())
```

Comment:
Let's make the session key configurable, and may be use some other notation for system keys, like _csrf_token